### PR TITLE
Associations Project: Completion

### PR DESCRIPTION
**Pull request by Zibilyonik**
The views are extremely crude, but all the functions work as intended.

### CODE REVIEW

**Comment by Zibilyonik** [comment]
I've run into many issues while trying to implement an "Attend" button but as I've noticed that it is not a step on OdinProject site I've decided not to do so.

**Comment by Zibilyonik** [comment]
> I've run into many issues while trying to implement an "Attend" button but as I've noticed that it is not a step on OdinProject site I've decided not to do so.

The issue was fixed and the buttons were added.

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Changes Required :muscle: 
Hi there @Zibilyonik 
Great work in this project :clap: :+1: 
But before moving on there are some improvements to be made.

- [x] The linter is missing from your project. You also need to set up the stickler in the repo. Check the instructions [here](https://github.com/microverseinc/linters-config/tree/master/ror).
Screenshot from 2020-06-05 11-40-14

- [x] It's best practice to summarize all changes made in the PR description. That's is important because it makes your repo looks professional. So, you can edit the PR description and add summarize all changes made. Check the instructions [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/).
Screenshot from 2020-06-05 11-44-16


- [x] There is a bug that is showing a future event in past events
Screenshot from 2020-06-05 11-16-15

- [x] I notice you added a link for the user destroy itself but it's not working, It throws an error. Since it's not a requirement you can remove the link.

Please kindly make these changes and submit for code review again.
Happy coding :computer: 
**Sergio Torres**

File: db/schema.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+# be faster and is potentially less error prone than running all of your
+# migrations from scratch. Old migrations may fail to apply correctly if those
+# migrations use external dependencies or application code.
+#
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema.define(version: 2020_06_03_212021) do
+
+  create_table "attendances", force: :cascade do |t|
+    t.integer "user_id"
```
Comment (euqueme) [inline db/schema.rb:16]: #### Migrations

* [x] events table has creator_id as the foreign key ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

- Even though this won't change the functionality of your project, this was required to practice the use of foreign keys, please add the migration and correct your model as well

File: app/views/events/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+<% end %>
+
+<% if logged_in? %>
+  <% if @event.attendees.include? current_user %>
+    <%= button_to 'Withdraw', withdraw_event_path(@event), method: :post %>
+  <% else %>
+    <%= button_to 'Attend', attend_event_path(@event), method: :post %>
+  <% end %>
+<% end %>
```
Comment (euqueme) [inline app/views/events/show.html.erb]: * [x] There should be absolutely NO logic in views.

  > Views are only for displaying information that became available to it from its corresponding controller not generating it. **Resources**
* In [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) explains how to put if/else statements within helpers so let's do the same for your code. [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

File: app/views/shared/_header.html.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+
+<% if logged_in? %>
+    <h1>What's Up, <%= link_to "#{current_user.name}", "/users/#{current_user.id}" %></h1>
+    <%= link_to "Logout", '/logout', method: :destroy%>
+<%else%>
+    <%= button_to "Login", '/login', method: :get%>
+    <%= button_to "Sign Up", '/users/new', method: :get%>
+<%end%>
```
Comment (euqueme) [inline app/views/shared/_header.html.erb]: * [x] There should be absolutely NO logic in views.

  > Views are only for displaying information that became available to it from its corresponding controller not generating it. **Resources**
* In [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) explains how to put if/else statements within helpers so let's do the same for your code. [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

**Review by euqueme — CHANGES_REQUESTED** [review]
### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @Zibilyonik :balloon: 

### Thank you for doing the changes requested by the previous reviewer, your project doesn't have the faulty functionality anymore and all the requested behavior is working properly :medal_sports: :ok_hand: there are some changes to make still that wasn't noticed by the previous reviewer but yet are important to cover the project minimum requirements, so let's work on them :construction: :construction_worker: 

#### General

* [x] The project has a descriptive README.md as you know this is a requirement for all microverse projects so let's add it ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

#### Associations

##### User as event creator

I've left a comment in your code where this change should be added :point_down: 

#### Views & Helpers

I've left some changes that need to be done to two of your views in the comments below :arrow_down_small: 

#### Test
Make sure you cover at least the following test cases: and explain in your Readme how to execute them

* [x] models have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
* [x] authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
* [x] event managment features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

### That's it, :raised_hands: 

### Please :pray: do the changes :technologist: and submit back for a code review when you are done :memo: 

Happy coding! :rocket: 

Any question? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :vertical_traffic_light:

File: db/schema.rb (hunk 1 above)
Comment (Zibilyonik) [reply db/schema.rb:16]: The creator of the events has already been renamed to "Creator" in the event model. At no point in the OdinProject description has it been specified that it should be renamed in the table itself.

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,
Thank you for the improvements done
However, there are still changes to do

1. I got this error when running test
```rb
$ rspec
../home/maelfosso/.rvm/gems/ruby-2.7.1-rvm/gems/sprockets-4.0.0/lib/sprockets/base.rb:119: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/home/maelfosso/.rvm/gems/ruby-2.7.1-rvm/gems/sprockets-4.0.0/lib/sprockets/base.rb:78: warning: The called method `find_asset' is defined here
.......F

Failures:

  1) User should save with all validations provided
     Failure/Error: expect(user).to be_valid
       expected #<User id: nil, name: "Potato", email: "pot@ato.com", created_at: nil, updated_at: nil> to be valid, but got errors: Email has already been taken
     # ./spec/models/user_spec.rb:15:in `block (2 levels) in <main>'

Finished in 2.71 seconds (files took 2.28 seconds to load)
10 examples, 1 failure

Failed examples:

rspec ./spec/models/user_spec.rb:13 # User should save with all validations provided
```
Kindly fix that.

2. As indicated [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png), the integration test should also be written for even managements. Kindly add it

3. Please, use [`shoulda-matcher`](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) for testing models

Kindly make the changes and submit a new code review request
Thanks

Mael FOSSO

**Comment by uryelah** [comment]
Hi **Onur Sürmeli**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job fixing the issues with your tests!

-------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**

----

- [x] [OPTIONAL] For the next projects remember to Capitalize your commits.
Read more at : [https://chris.beams.io/posts/git-commit/#capitalize](https://chris.beams.io/posts/git-commit/#capitalize)

---------

    
    
**Keep being an awesome dev and good luck with your next project!**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Pull request merged**